src/app/services/office.service.ts

```diff
@@ -618,48 +639,50 @@ export class OfficeService {
     });
   }
 
-  /**
-   * Versteckt einen Range, indem der Text unsichtbar gesetzt wird.
-   */
-  async hideRange(range: Word.Range): Promise<Word.Range> {
+  async hideRange(range: Word.Range): Promise<void> {
     return Word.run(range, context => {
       const ooxml = range.getOoxml();
 
       return context.sync().then(() => {
         const parser = new DOMParser();
         const doc = parser.parseFromString(ooxml.value, 'application/xml');
-
-        const el = doc.getElementsByTagName('w:t');
-
-        for (let i = 0; i < el.length; i++) {
-          const t = el.item(i);
-          const vanish = doc.createElementNS('http://schemas.openxmlformats.org/wordprocessingml/2006/main', 'w:vanish');
-          let rpr = t.previousSibling;
-
-          if (!rpr) {
-            rpr = doc.createElementNS('http://schemas.openxmlformats.org/wordprocessingml/2006/main', 'w:rPr');
-            t.parentNode.insertBefore(rpr, t);
+        const body = doc.getElementsByTagName('w:body').item(0);
+        const children = Array.from(body.childNodes).filter(it => ['w:sdt', 'w:p', 'w:tbl'].indexOf(it.nodeName) !== -1);
+
+        children.map(it => {
+          switch (it.nodeName) {
+            case 'w:sdt': {
```

Comment:
Es gibt 5 hide Methoden aber in dem switch case sind nur 3 Fälle behandelt.

Reply:
OOXML ist eine Baumstruktur und an oberster Stelle kommen eben nur drei der Elemente vor. Die anderen Funktionen beziehen sich auf tieferliegende Elemente.
http://www.datypic.com/sc/ooxml/e-w_body-1.html